libraries/Microsoft.Bot.Builder/IBotTelemetryClientExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Bot.Builder
+{
+    public static class IBotTelemetryClientExtensions
+    {
+        /// <summary>
+        /// Adds the ability to call TrackPageView on the TelemetryClient if it implements IBotPageViewTelemetryClient as well as IBotTelemetryClient.
+        /// </summary>
+        /// <param name="telemetryClient">The TelemetryClient that implements IBotTelemetryClient.</param>
+        /// <param name="dialogName">The name of the dialog to log the entry / start for.</param>
+        /// <param name="properties">Named string values you can use to search and classify events.</param>
+        /// <param name="metrics">Measurements associated with this event.</param>
+        public static void TrackPageView(this IBotTelemetryClient telemetryClient, string dialogName, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null)
```

Comment:
why not call this extension method TrackDialog? isn't that what you original intended?

Reply:
I though we had agreed to fall back to the more native event names. Given this is an extension to and therefore used like it is on IBotTelemetryClient, then we should name it however we would name it if we were just extending the interface.

Reply:
I was seeing this as layered. Lower layer being derived directly from the raw capabilities of the service we are abstracting. This lower layer is captured in a series of interfaces. (Used to be one interface, now it's two.)

The higher layer is more about the usage pattern in the application. For example, TrackDialog corresponds to the concept of Dialogs that exist only with the application and not the telemetry infrastructure. The implementation of the provider happens to map this to a PageView because we felt that sort of fitted OK.

The benefit of the multiple interfaces was that this application concept of Dialog could actually be mapped differently in a less capable (aka legacy) telemetry provider. This negotiation (implemented with a runtime type check) facilitates the back compat and graceful fallback.

Anyhow the abstractions TrackDialog and TrackPageView target different audiences. The TRackPageView being intended for someone implementing something like TrackDialog.

Anyhow that was my rational. I didn't see the extension method as extending the interface, more as a higher level abstraction, as if we had always intended it to work this way.

Reply:
Ok - I am fine either way and do see your side of the argument too. I will alter the naming. Ironically, that is how I had named it when this PR first got opened :)

Reply:
Updated extension method to DialogView as discussed.